openhands/server/routes/new_conversation.py

```diff
@@ -51,10 +49,10 @@ async def new_conversation(request: Request, data: InitSessionRequest):
 
     conversation_store = await ConversationStore.get_instance(config)
 
-    conversation_id = uuid.uuid4().hex
+    conversation_id = sio.eio.generate_id()
     while await conversation_store.exists(conversation_id):
         logger.warning(f'Collision on conversation ID: {conversation_id}. Retrying...')
-        conversation_id = uuid.uuid4().hex
+        conversation_id = sio.eio.generate_id()
```

Comment:
I'm leaving this in right now even though I don't think it is necessary - with a 128 bit number you could try every second until the sun burns out and not hit a duplicate.